packages/ace-mode/index.js

```diff
@@ -1,266 +1,23 @@
-  '$max',
-  '$min',
-  '$push',
-  '$stdDevPop',
-  '$stdDevSamp',
-  '$sum'
-];
-
-var AGG_CONVERTERS = [
-  '$convert',
-  '$ltrim',
-  '$rtrim',
-  '$toBool',
-  '$toDate',
-  '$toDecimal',
-  '$toDouble',
-  '$toInt',
-  '$toLong',
-  '$toObjectId',
-  '$toString',
-  '$trim'
-];
-
-var AGG_EXPRESSION_OPERATORS = [
-  '$abs',
-  '$add',
-  '$allElementsTrue',
-  '$and',
-  '$anyElementTrue',
-  '$arrayElemAt',
-  '$arrayToObject',
-  '$ceil',
-  '$cmp',
-  '$concat',
-  '$concatArrays',
-  '$cond',
-  '$dateFromParts',
-  '$dateFromString',
-  '$dateToParts',
-  '$dateToString',
-  '$dayOfMonth',
-  '$dayOfWeek',
-  '$dayOfYear',
-  '$dateToString',
-  '$divide',
-  '$eq',
-  '$exp',
-  '$filter',
-  '$floor',
-  '$gt',
-  '$gte',
-  '$hour',
-  '$ifNull',
-  '$in',
-  '$indexOfArray',
-  '$indexOfBytes',
-  '$indexOfCP',
-  '$isArray',
-  '$isoDayOfWeek',
-  '$isoWeek',
-  '$isoWeekYear',
-  '$let',
-  '$literal',
-  '$lt',
-  '$lt',
-  '$ln',
-  '$log',
-  '$log10',
-  '$map',
-  '$mergeObjects',
-  '$meta',
-  '$millisecond',
-  '$minute',
-  '$mod',
-  '$month',
-  '$multiply',
-  '$new',
-  '$not',
-  '$objectToArray',
-  '$or',
-  '$pow',
-  '$range',
-  '$reduce',
-  '$reverseArray',
-  '$second',
-  '$setDifference',
-  '$setEquals',
-  '$setIntersection',
-  '$setIsSubset',
-  '$setUnion',
-  '$size',
-  '$slice',
-  '$split',
-  '$sqrt',
-  '$strcasecmp',
-  '$strLenBytes',
-  '$strLenCP',
-  '$substr',
-  '$substrBytes',
-  '$substrCP',
-  '$subtract',
-  '$switch',
-  '$toLower',
-  '$toUpper',
-  '$trunc',
-  '$type',
-  '$week',
-  '$year',
-  '$zip'
-];
-
-var AGG_QUERY_OPERATORS = [
-  '$all',
-  '$and',
-  '$bitsAllClear',
-  '$bitsAllSet',
-  '$bitsAnyClear',
-  '$bitsAnySet',
-  '$comment',
-  '$elemMatch',
-  '$eq',
-  '$exists',
-  '$expr',
-  '$geoIntersects',
-  '$geoWithin',
-  '$gt',
-  '$gte',
-  '$in',
-  '$jsonSchema',
-  '$lt',
-  '$lte',
-  '$mod',
-  '$ne',
-  '$near',
-  '$nearSphere',
-  '$nin',
-  '$not',
-  '$nor',
-  '$or',
-  '$regex',
-  '$size',
-  '$slice',
-  '$text',
-  '$type',
-  '$where'
-];
-
-var AGG_STAGE_OPERATORS = [
-  '$addFields',
-  '$bucket',
-  '$bucketAuto',
-  '$collStats',
-  '$count',
-  '$facet',
-  '$geoNear',
-  '$graphLookup',
-  '$group',
-  '$indexStats',
-  '$limit',
-  '$lookup',
-  '$match',
-  '$out',
-  '$project',
-  '$redact',
-  '$replaceRoot',
-  '$sample',
-  '$skip',
-  '$sort',
-  '$sortByCount',
-  '$unwind'
-];
-
-var EXT_JSON = [
-  '$binary',
-  '$date',
-  '$maxKey',
-  '$minKey',
-  '$numberDecimal',
-  '$numberDouble',
-  '$numberInt',
-  '$numberLong',
-  '$oid',
-  '$regex',
-  '$regularExpression',
-  '$timestamp',
-  '$undefined'
-];
-
-var SUPPORT = [];
-
-SUPPORT.push.apply(SUPPORT, AGG_ACCUMULATORS);
-SUPPORT.push.apply(SUPPORT, AGG_CONVERTERS);
-SUPPORT.push.apply(SUPPORT, AGG_EXPRESSION_OPERATORS);
-SUPPORT.push.apply(SUPPORT, AGG_QUERY_OPERATORS);
-SUPPORT.push.apply(SUPPORT, AGG_STAGE_OPERATORS);
-SUPPORT.push.apply(SUPPORT, EXT_JSON);
+const {
```

Comment:
Do we still want this to be a package? We don't use it elsewhere internally as far as I understand, would be nice to have this closer to the rest of the "Editor" component

Reply:
I think that I would slightly prefer to keep all the ace stuff in one separate package inside monorepo, inlcuding themes and a worker eventually. I think it would be nice if we can utilize webpack to bundle some of these files correctly and this would require using webpack-specific imports most definitely (like importing workers seems to require using source-loader, or instead of maintaining a theme in a unformatted, not highlighted string inside javascript, I would prefer to have it in a css file that I can import with the style loader or source-loader) and so I would prefer to keep all this non-standard stuff away from compass-components. But it's a minor thing, tell me if you think that this is not a good enough reason to do so

Reply:
Ugh, actually a theme will probably require us to depend on leafygreen / compass-components here anyway, hmm... I guess we can switch compass-components to be a webpack bundled package instead of just a typescript build then as an alternative. Any preference?

Reply:
I'll postpone the decision on this for another patch if you don't mind. We definitely can always merge it, I would just consider the implication of this if we decide to